ktor-features/ktor-webjars/src/io/ktor/webjars/Webjars.kt

```diff
@@ -0,0 +1,85 @@
+package io.ktor.webjars
+
+import io.ktor.application.ApplicationCall
+import io.ktor.application.ApplicationCallPipeline
+import io.ktor.application.ApplicationFeature
+import io.ktor.application.call
+import io.ktor.http.ContentType
+import io.ktor.http.HttpMethod
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.defaultForFilePath
+import io.ktor.pipeline.PipelineContext
+import io.ktor.request.httpMethod
+import io.ktor.request.uri
+import io.ktor.response.respond
+import io.ktor.response.respondBytes
+import io.ktor.util.AttributeKey
+import org.webjars.MultipleMatchesException
+import org.webjars.WebJarAssetLocator
+import java.nio.file.Paths
+
+class Webjars(val configuration: Configuration) {
+
+    private fun fileName(path: String) : String = Paths.get(path).fileName.toString()
+
+    private fun extractWebJar(path : String) : String {
+        val firstSlash = if (path.startsWith("/")) 1 else 0
+        val nextSlash = path.indexOf("/", 1)
+        val webjar = if(nextSlash > -1) path.substring(firstSlash,  nextSlash) else ""
+        val partialPath = path.substring(nextSlash + 1)
+        return locator.getFullPath(webjar, partialPath)
+    }
+
+    private val locator = WebJarAssetLocator()
```

Comment:
Is is stateless? Is is thread-safe? Is it safe to call `getFullPath` concurrently?

Reply:
From AFAIK yes, it's thread safe, and can be called concurrently. Spring would be on pretty bad grounds if this were not thread-safe ;)